src/Diagnostics.ModelsAndUtils/Models/ResponseExtensions/Solution.cs

```diff
@@ -8,123 +8,96 @@ namespace Diagnostics.ModelsAndUtils.Models.ResponseExtensions
 
         /// <summary>
-        /// If the solution requires confirmation, it will signify a potentially dangerous action in the UI.
+        /// Options pertaining to <see cref="ActionType.ArmApi"/>.
+        /// This is ignored if <see cref="Action"/> is not <see cref="ActionType.ArmApi"/>.
         /// </summary>
-        public bool RequiresConfirmation { get; set; }
+        public ArmApiOptions ApiOptions { get; set; }
 
         /// <summary>
-        /// The URI of the target resource on which an action will be performed.
+        /// Options pertaining to <see cref="ActionType.GoToBlade"/>.
+        /// This is ignored if <see cref="Action"/> is not <see cref="ActionType.GoToBlade"/>.
         /// </summary>
-        public string ResourceUri { get; set; }
+        public GoToBladeOptions BladeOptions { get; set; }
 
         /// <summary>
-        /// Reads <see cref="OperationContext"/> to determine if an internal view should be rendered.
+        /// Options pertaining to <see cref="ActionType.OpenTab"/>.
+        /// This is ignored if <see cref="Action"/> is not <see cref="ActionType.OpenTab"/>.
         /// </summary>
-        public bool IsInternal { get; set; } = false;
+        public OpenTabOptions TabOptions { get; set; }
 
         /// <summary>
-        /// Instructions that will be sent to the customer by support staff. Rendered in Markdown.
+        /// Free-form JSON-serializable options that will override <see cref="ApiOptions"/>,
+        /// <see cref="BladeOptions"/>, or <see cref="TabOptions"/> for the purpose of compatibility.
+        /// Prefer the aforementioned first-class option objects.
         /// </summary>
-        public string InternalInstructions { get; set; }
+        public Dictionary<string, object> OverrideOptions { get; set; }
 
         /// <summary>
-        /// A direct link to the detector that is added to the instructions for the customer.
+        /// If the solution requires confirmation, it will signify a potentially dangerous action in the UI.
         /// </summary>
-        public Uri DetectorLink { get; set; }
+        public bool RequiresConfirmation { get; set; }
 
         /// <summary>
-        /// Denotes which action will be performed, such as a specific API call or navigation link.
+        /// The URI of the target resource on which an action will be performed.
         /// </summary>
-        [JsonConverter(typeof(StringEnumConverter))]
-        public ActionType Action { get; set; }
+        public string ResourceUri { get; set; }
 
         /// <summary>
-        /// Free-form JSON-serializable arguments that can be sent as the body of the action request if a request
-        /// body is applicable.
+        /// Instructions that will be sent to the customer by support staff. Rendered as Markdown.
         /// </summary>
-        public Dictionary<string, object> ActionArgs { get; set; }
+        public string InternalInstructions { get; set; }
```

Comment:
>InternalInstructions [](start = 22, length = 20)

Calling this InternalMarkdown makes it explicit that you can pass Markdown to this property